sets/slices.go

```diff
@@ -58,24 +58,16 @@ func IntersectStable[T comparable](slices ...[]T) []T {
 	if len(slices) == 0 {
 		return []T{}
 	}
-
-	// Make a copy of the first slice; we're going to gradually remove elements.
-	final := append([]T{}, slices[0]...)
-
-	// Remove duplicates
-	finalSeen := make(map[T]struct{}, len(final))
-	var i int
-	for _, v := range final {
-		if _, ok := finalSeen[v]; !ok {
-			finalSeen[v] = struct{}{}
-			final[i] = v
-			i++
+	// Make a copy of the slice with duplicates removed.
+	final := make([]T, 0, len(slices[0]))
```

Comment:
If you defer allocation until after finding uniq, you can size the slice to the exact size you need

Reply:
Wouldn't that require another full iteration over `uniq`?

Reply:
You're right, it would